drivers/flash/flash_stm32l5x.c

```diff
@@ -0,0 +1,381 @@
+/*
+ * Copyright (c) 2021 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_DOMAIN flash_stm32l5
+#define LOG_LEVEL CONFIG_FLASH_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <drivers/flash.h>
+#include <init.h>
+#include <soc.h>
+#include <stm32l5xx_ll_icache.h>
+#include <stm32_ll_system.h>
+
+#include "flash_stm32.h"
+
+#define STM32L5_SERIES_MAX_FLASH	512
+#define BANK2_OFFSET	(KB(STM32L5_SERIES_MAX_FLASH) / 2)
+
+#define ICACHE_DISABLE_TIMEOUT_VALUE           1U   /* 1ms */
+#define ICACHE_INVALIDATE_TIMEOUT_VALUE        1U   /* 1ms */
+
+static void stm32l5x_icache_clear_status(void)
+{
```

Comment:
We cannot just clear the flags like that, it is necessary to take the appropriate action corresponding.
for example if flash is busy, we may need to wait, or cancel the previous action ...
if there is an error, should we report the error, or reset the IP to recover ...